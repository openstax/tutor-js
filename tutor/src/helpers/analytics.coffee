_ = require 'underscore'

Routes = require '../routes'

{CurrentUserStore} = require '../flux/current-user'

DestinationHelper  = require '../helpers/routes-and-destinations'

# generate custom event data for routes
Events =
  viewTaskStep: ({courseId}) ->
    # track all work done on a course
    Analytics.sendEvent 'Course', 'Work', label: courseId

  viewStudentDashboard: ({courseId}) ->
    # compare activity between courses
    Analytics.sendEvent 'Student', 'Dashboard', label: courseId


# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
Translators =

  dashboard:            ({courseId}) -> "/student/choose-course/#{courseId}"
  viewPractice:         ({courseId}) -> "/student/practice/#{courseId}"
  viewPerformanceForecast:  ({courseId}) -> "/student/performance-forecast/#{courseId}"
  viewStudentDashboard: ({courseId}) -> "/student/dashboard/#{courseId}"

  calendarByDate:        ({courseId}) -> "/teacher/calendar/#{courseId}"
  viewTeacherPerformanceForecast:      ({courseId}) -> "/teacher/performance-forecast/#{courseId}"
  viewScores:            ({courseId}) -> "/teacher/student-scores/#{courseId}"
  courseSettings:        ({courseId}) -> "/teacher/roster/#{courseId}"
  editReading:           ({courseId}) -> "/teacher/assignment/edit/reading/#{courseId}"
  editHomework:          ({courseId}) -> "/teacher/assignment/edit/homework/#{courseId}"
  editExternal:          ({courseId}) -> "/teacher/assignment/edit/external/#{courseId}"
  createReading:         ({courseId}) -> "/teacher/assignment/create/reading/#{courseId}"
  createHomework:        ({courseId}) -> "/teacher/assignment/create/homework/#{courseId}"
  createExternal:        ({courseId}) -> "/teacher/assignment/create/external/#{courseId}"
  calendarViewPlanStats: ({courseId}) -> "/teacher/metrics/quick/#{courseId}"
  reviewTask:            ({courseId}) -> "/teacher/metrics/review/#{courseId}"

  viewReferenceBook:        ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookSection: ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookPage:    ({courseId}) -> "/reference-view/#{courseId}"

  # Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep:         ({courseId}) ->
    role = CurrentUserStore.getCourseRole(courseId, true)
    "/#{role}/task-step/#{courseId}"


GA = undefined

Analytics =

  setTracker: (tracker) -> GA = tracker

  sendPageView: (url) ->
    GA?('send', 'pageview', url)

  onNavigation: (path) ->
    return unless GA

    route = Routes.pathToEntry(path)

    return @sendPageView("/not-found/#{change.path}") unless route

    path = Translators[route.entry.name]?( route.match.params ) or route.match.pathname

    # if we're also going to send custom events then we set the page
    if Events[route.entry.name]
      GA('set', 'page', path)
      Events[route.entry.name]( route.match.params )
      @sendPageView() # url's not needed since it was set before events
    else
      @sendPageView(route.match.pathname)

  sendEvent: (category, action, attrs) ->
    return unless GA
    GA('send', 'event', category, action, attrs.label, attrs.value)

module.exports = Analytics
