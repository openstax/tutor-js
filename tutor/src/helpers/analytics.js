import { partial, get } from 'lodash';

import Router from '../helpers/router';
import Courses from '../models/courses-map';

// generate custom event data for routes
const Events = {
  viewTaskStep({ courseId }) {
    // track all work done on a course
    return Analytics.sendEvent('Course', 'Work', { label: courseId });
  },

  viewStudentDashboard({ courseId }) {
    // compare activity between courses
    return Analytics.sendEvent('Student', 'Dashboard', { label: courseId });
  },
};

// a bit shorter helper methods
const isTeacher = courseId => get(Courses.get(courseId), 'currentRole.isTeacher', false);
const getRole = function(courseId) {
  if (Courses.get(courseId).is_teacher) { return 'teacher'; } else { return 'student'; }
};

const assignmentTypeTranslator = function(assignmentType, { courseId, id }) {
  const newOrEdit = id === 'new' ? 'create' : 'edit';
  return `/teacher/assignment/${newOrEdit}/${assignmentType}/${courseId}`;
};

function viewReferenceBook({ courseId, pageId }) {
  const url = `/reference-view/${courseId}`;
  if (!pageId) {
    return url;
  }
  const course = Courses.get(courseId);
  const page = course.referenceBook.pages.byId.get(pageId);
  return page ? `${url}/section/${page.chapter_section.key}` : url;
}

// Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
const Translators = {
  dashboard({ courseId }) {
    if (isTeacher(courseId)) { return `/teacher/calendar/${courseId}`; } else { return `/student/dashboard/${courseId}`; }
  },
  practiceTopics({ courseId }) { return `/student/practice/${courseId}`; },
  viewPerformanceGuide({ courseId }) {
    return `/${getRole(courseId)}/performance-forecast/${courseId}`;
  },
  calendarByDate({ courseId }) { return `/teacher/calendar/${courseId}`; },
  viewGradebook({ courseId }) { return `/teacher/student-scores/${courseId}`; },
  courseSettings({ courseId }) { return `/teacher/roster/${courseId}`; },
  editAssignment: partial(assignmentTypeTranslator, 'assignment'),
  calendarViewPlanStats({ courseId }) { return `/teacher/metrics/quick/${courseId}`; },
  reviewTask({ courseId }) { return `/teacher/metrics/review/${courseId}`; },
  viewReferenceBookSection: viewReferenceBook,
  viewReferenceBookPage: viewReferenceBook,
  viewReferenceBook,

  // Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep({ courseId }) {
    const course = Courses.get(courseId);
    const role = course ? course.primaryRole.type : 'unknown';
    return `/${role}/task-step/${courseId}`;
  },
};

const TAG_MANAGER_ID = 'GTM-W6N7PB';

let GA = undefined;
let trackerNames = undefined;

var Analytics = {
  // copied from BIT
  initTagManager(w, d, s, l, i) {
    // Disable ESLint rules since we're copying Google's script
    /* eslint-disable one-var, prefer-const, prefer-template */
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    let f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l !== 'dataLayer' ? '&l=' + l : '';
    // Breaks in tests because there are no scripts
    if (f) {
      j.async = true;
      j.src = '//www.googletagmanager.com/gtm.js?id=' + i + dl;
      f.parentNode.insertBefore(j, f);
    }
  },
  
  // Set the Command Queue (ga) and cache the (possibly multiple) tracker names
  setGa(ga) {
    trackerNames = undefined;
    GA = ga;
    if (GA) {
      GA(function() {
        trackerNames = GA.getAll().map(function(tracker) { return tracker.get('name'); });
      });
      this.initTagManager(window, document, 'script', 'dataLayer', TAG_MANAGER_ID);
    }

    return GA;
  },

  // Calls all trackers with the command and arguments given here
  ga(command, ...params) {
    if (!GA || !trackerNames) { return; }
    trackerNames.forEach((trackerName) => {
      this.realGa(trackerName + '.' + command, ...params);
    });
  },

  // Exists largely for spec mocking
  realGa(...args) {
    GA(...args);
  },

  sendPageView(url) {
    this.ga('send', 'pageview', url);
  },

  recordCourseDimension(courseId) {
    const course = Courses.get(courseId);
    if (course) {
      this.ga('set', 'dimension1', course.appearance_code);
    }
  },

  onNavigation(path) {
    if (!GA) { return; }
    const route = Router.currentMatch(path);

    if (!route) {
      this.sendPageView(`/not-found${path}`);
      return;
    }
    const { courseId } = route.params;
    if (courseId) { this.recordCourseDimension(courseId); }
    const translatedPath = Translators[route.entry.name] ?
      Translators[route.entry.name]( route.params ) : route.pathname;

    this.ga('set', 'page', translatedPath);
    // if we're also going to send custom events then we set the page
    if (Events[route.entry.name]) {
      Events[route.entry.name]( route.params );
    }

    this.sendPageView(); // url's not needed since it was set before events
  },

  sendEvent(category, action, attrs = {}) {
    this.ga('send', 'event', category, action, attrs.label, attrs.value);
  },
};

export default Analytics;
