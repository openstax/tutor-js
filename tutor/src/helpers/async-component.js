import React from 'react';
import AsyncErrorBoundary from '../components/error-monitoring/async-load-error';
import LoadingAnimation from '../../../shared/src/components/loading-animation'

export function asyncComponent(loader, name = '') {

    const Component = React.lazy(loader);
    const loading = <LoadingAnimation message={`Loading ${name}â€¦`} />;

    const Loader = (props) => (
        <AsyncErrorBoundary>
            <React.Suspense fallback={loading}>
                <Component {...props} />
            </React.Suspense>
        </AsyncErrorBoundary>
    );

    return (props) => <Loader {...props} />;
}


export function loadAsync(resolve, name) {
    // return a function so the router will only evaluate it when it's needed
    // in the future we may insert role dependant logic here
    return () => asyncComponent(resolve, name);
}
