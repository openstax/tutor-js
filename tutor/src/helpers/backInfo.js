import { navHistory } from '../models'


const backInfo = () => {
    // Gets route to last path that was not the same as the current one
    // See TransitionStore for more detail.
    const historyInfo = navHistory.latest

    const text = historyInfo != null && !!historyInfo.name ? `Back to ${historyInfo.name}` : 'My Courses';
    const to =  historyInfo != null && !!historyInfo.path ? historyInfo.path : 'myCourses';

    return {
        text,
        to,
    };
};


export default backInfo;
