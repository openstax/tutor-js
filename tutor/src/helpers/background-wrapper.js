import { styled } from 'vendor';
import { navbars } from '../theme.js';

const BackgroundWrapper = styled.div`
  background: #fff;
  min-height: calc(100vh - ${navbars.top.height} - ${navbars.bottom.height});
  position: relative;
  overflow: hidden;
  padding: 0 2.4rem;
`;

const ContentWrapper = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

export { BackgroundWrapper, ContentWrapper };
