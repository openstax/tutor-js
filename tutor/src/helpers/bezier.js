function EASE_IN_OUT(t) {
    if (t < .5) { return 4 * t * t * t; } else { return ((t - 1) * ((2 * t) - 2) * ((2 * t) - 2)) + 1; }
}

function CALC(start, end, elapsed, duration) {
    if (elapsed > duration) { return end; }
    return start + ((end - start) * EASE_IN_OUT(elapsed / duration));
}

export default function bezier({
    range, duration, onStep, onComplete,
}) {
    const startTime = Date.now();
    const requestAnimationFrame = window.requestAnimationFrame || window.setTimeout;

    const step = () => {
        const elapsed = Date.now() - startTime;
        onStep(CALC(range[0], range[1], elapsed, duration));
        if (elapsed < duration) {
            requestAnimationFrame(step);
        } else if (onComplete) {
            onComplete();
        }
    };

    step();
}
