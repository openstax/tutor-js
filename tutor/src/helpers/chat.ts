import { Raven } from '../models'

const DEPLOYMENT_ID   = '5724P000000ghju';
const ORGANIZATION_ID = '00DU0000000Kwch';
const ELEMENT_ID      = '5736f000000k9eQ';


const win = window as any

export const Chat = {
    initialized: false,

    get isEnabled() {
        return Boolean(win.liveagent);
    },

    get isOnline() {
        if (!this.isEnabled) { return false; }
        const chat = document.querySelector<HTMLDivElement>('.chat.enabled');
        return chat && chat.style.display !== 'none';
    },

    initialize() {
    // will emit debugging info to the console
    // liveagent.enableLogging();
    },

    setElementVisiblity(online: any, offline?: any) {
        if (!this.isEnabled) { return; }
        try {
            win.liveagent.showWhenOnline(ELEMENT_ID, online);
            if (offline) {
                win.liveagent.showWhenOffline(ELEMENT_ID, offline);
            }
            if (!this.initialized) {
                win.liveagent.init('https://d.la1-c2-ia5.salesforceliveagent.com/chat', DEPLOYMENT_ID, ORGANIZATION_ID);
                this.initialized = true;
            }
        } catch(err) {
            Raven.captureException(err);
        }
    },

    start() {
        win.liveagent.startChat(ELEMENT_ID);
    },

};
