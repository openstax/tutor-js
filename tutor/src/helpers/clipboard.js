import _ from 'underscore';

const SUPPORTED = typeof document.queryCommandSupported === 'function' ? document.queryCommandSupported('copy') : undefined;

export default {
    isSupported: _.constant(!!SUPPORTED),

    copy() {
        if (!SUPPORTED) { return; }

        try {
            document.execCommand('copy');
        } catch (e) {
            // eslint-disable-next-line no-console
            console.warn('clipboard copy failed', e);
        }
    },
};
