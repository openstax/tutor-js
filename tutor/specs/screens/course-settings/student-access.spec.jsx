import StudentAccess from '../../../src/screens/course-settings/student-access';
import { ApiMock, Factory } from '../../helpers'

jest.mock('../../../src/helpers/clipboard');

describe('Course Settings, student access', () => {

    let props;
    let course;

    ApiMock.intercept({
        'lms/courses/\\d+': () => ({
            key: 'key-1234',
            secret: 'secret-1234',
            launch_url: 'launch-url',
            configuration_url: 'config-url-1234',
        }),
    })

    beforeEach(() => {
        course = Factory.course()
        props = {
            course,
        };
    });

    it('renders with only link for non-lms enabled', () => {
        const access = mount(<StudentAccess {...props} />);
        expect(access).toHaveRendered(
            `.direct-links-only input[value="${course.periods[0].enrollment_url_with_details}"]`,
        );
        expect.snapshot(<StudentAccess {...props} />).toMatchSnapshot();
    });

    it('renders chooser when lms is enabled', () => {
        course.is_lms_enabling_allowed = true;
        expect.snapshot(<StudentAccess {...props} />).toMatchSnapshot();
    });

    it('can switch to links', () => {
        course.is_lms_enabled = true
        course.is_lms_enabling_allowed = true;
        course.save = jest.fn();
        const access = mount(<StudentAccess {...props} />);
        access.find('.choice.links').simulate('click');
        expect(course.save).not.toHaveBeenCalled();
        document.querySelector('.modal.warn-before-links .btn-primary').click();
        expect(course.save).toHaveBeenCalled();
        expect(course.is_lms_enabled).toEqual(false);
        expect.snapshot(<StudentAccess {...props} />).toMatchSnapshot();
        access.unmount();
    });

    it('can switch to lms', () => {
        course.is_lms_enabled = false
        course.is_lms_enabling_allowed = true;
        course.save = jest.fn();
        course.lms.key = 'key-123';
        course.lms.secret = 'secret-sshhh';
        course.lms.url = 'test.url.com';
        course.lms.fetch = jest.fn();
        const access = mount(<StudentAccess {...props} />);
        access.find('.choice.lms').simulate('click');
        expect(course.lms.fetch).toHaveBeenCalled();
        expect(course.save).toHaveBeenCalled();
        expect(course.is_lms_enabled).toEqual(true);
        expect.snapshot(<StudentAccess {...props} />).toMatchSnapshot();
        access.unmount();
    });
});
