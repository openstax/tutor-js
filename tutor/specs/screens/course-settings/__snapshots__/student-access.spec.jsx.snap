// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Settings, student access can switch to links 1`] = `
.c0 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="student-access"
>
  <p>
    Choose how students access OpenStax Tutor. Access settings cannot be changed after students begin to enroll.
  </p>
  <a
    href="https://openstax.secure.force.com/help/articles/FAQ/Student-Access-with-Enrollment-Link-versus-LMS-sign-in-for-OpenStax-Tutor-Beta-LMS-Pilots"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-info-circle fa-w-16 c0 ox-icon ox-icon-info-circle"
      data-icon="info-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     Which option is right for my course?
  </a>
  <div>
    <div
      className="links active card"
    >
      <div
        className="card-header"
      >
        <div
          aria-selected={true}
          className="links choice checked"
          onClick={[Function]}
          role="button"
        >
          <div
            className="heading"
          >
            <h3
              className="title"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check-square fa-w-14 c0 ox-icon ox-icon-check-square"
                color="#63a524"
                data-icon="check-square"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352zm-35.864-241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Give students direct links
            </h3>
            <p
              className="info"
            >
              You will give students links to access OpenStax Tutor directly.
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <p>
          Send your students their section's direct links to enroll.
        </p>
        <label
          className="copy-on-focus"
        >
          1st
          <input
            className={null}
            onFocus={[Function]}
            readOnly={true}
            value="http://test/period/1/Local-Test-Course-Two-Spring-2017"
          />
        </label>
      </div>
    </div>
    <div
      className="lms card"
    >
      <div
        className="card-header"
      >
        <div
          aria-selected={false}
          className="lms choice"
          onClick={[Function]}
          role="button"
        >
          <div
            className="heading"
          >
            <h3
              className="title"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-square fa-w-14 c0 ox-icon ox-icon-square"
                color="#929292"
                data-icon="square"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Integrate with your learning management system (LMS)
            </h3>
            <p
              className="info"
            >
              Integrate OpenStax Tutor with Blackboard, Canvas, etc., to send student course averages to your LMS and enable single sign on.
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      />
    </div>
  </div>
</div>
`;

exports[`Course Settings, student access can switch to lms 1`] = `
.c0 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="student-access"
>
  <p>
    Choose how students access OpenStax Tutor. Access settings cannot be changed after students begin to enroll.
  </p>
  <a
    href="https://openstax.secure.force.com/help/articles/FAQ/Student-Access-with-Enrollment-Link-versus-LMS-sign-in-for-OpenStax-Tutor-Beta-LMS-Pilots"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-info-circle fa-w-16 c0 ox-icon ox-icon-info-circle"
      data-icon="info-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     Which option is right for my course?
  </a>
  <div>
    <div
      className="links card"
    >
      <div
        className="card-header"
      >
        <div
          aria-selected={false}
          className="links choice"
          onClick={[Function]}
          role="button"
        >
          <div
            className="heading"
          >
            <h3
              className="title"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-square fa-w-14 c0 ox-icon ox-icon-square"
                color="#929292"
                data-icon="square"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Give students direct links
            </h3>
            <p
              className="info"
            >
              You will give students links to access OpenStax Tutor directly.
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <p>
          Send your students their section's direct links to enroll.
        </p>
        <label
          className="copy-on-focus"
        >
          1st
          <input
            className={null}
            onFocus={[Function]}
            readOnly={true}
            value="http://test/period/1/Local-Test-Course-Two-Spring-2017"
          />
        </label>
      </div>
    </div>
    <div
      className="lms active card"
    >
      <div
        className="card-header"
      >
        <div
          aria-selected={true}
          className="lms choice checked"
          onClick={[Function]}
          role="button"
        >
          <div
            className="heading"
          >
            <h3
              className="title"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check-square fa-w-14 c0 ox-icon ox-icon-check-square"
                color="#63a524"
                data-icon="check-square"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352zm-35.864-241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Integrate with your learning management system (LMS)
            </h3>
            <p
              className="info"
            >
              Integrate OpenStax Tutor with Blackboard, Canvas, etc., to send student course averages to your LMS and enable single sign on.
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="lms-access"
        >
          <div
            className="btn-group btn-group-small btn-group-toggle"
            role="group"
          >
            <label
              className="btn active btn-default"
              disabled={false}
            >
              <input
                autoComplete="off"
                checked={true}
                disabled={false}
                name="vendor"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="blackboard"
              />
              Blackboard
            </label>
            <label
              className="btn btn-default"
              disabled={false}
            >
              <input
                autoComplete="off"
                checked={false}
                disabled={false}
                name="vendor"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="canvas"
              />
              Canvas
            </label>
            <label
              className="btn btn-default"
              disabled={false}
            >
              <input
                autoComplete="off"
                checked={false}
                disabled={false}
                name="vendor"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="moodle"
              />
              Moodle
            </label>
            <label
              className="btn btn-default"
              disabled={false}
            >
              <input
                autoComplete="off"
                checked={false}
                disabled={false}
                name="vendor"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="d2l"
              />
              Desire2Learn
            </label>
            <label
              className="btn btn-default"
              disabled={false}
            >
              <input
                autoComplete="off"
                checked={false}
                disabled={false}
                name="vendor"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="sakai"
              />
              Sakai
            </label>
          </div>
          <p>
            Copy the information below and paste into your LMS where prompted. Then launch OpenStax Tutor from your LMS to pair.
          </p>
          <div
            className="blackboard"
          >
            <a
              className="external-icon"
              href="https://openstax.secure.force.com/help/articles/FAQ/Blackboard-LMS-integration-for-OpenStax-Tutor-Beta"
              tabIndex={0}
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-info-circle fa-w-16 c0 ox-icon ox-icon-info-circle"
                data-icon="info-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
               How do I integrate with Blackboard?
            </a>
            <label
              className="copy-on-focus"
            >
              URL
              <input
                className={null}
                onFocus={[Function]}
                readOnly={true}
              />
            </label>
            <label
              className="copy-on-focus"
            >
              Key
              <input
                className={null}
                onFocus={[Function]}
                readOnly={true}
                value="key-123"
              />
            </label>
            <label
              className="copy-on-focus"
            >
              Secret
              <input
                className={null}
                onFocus={[Function]}
                readOnly={true}
                value="secret-sshhh"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Course Settings, student access renders chooser when lms is enabled 1`] = `
.c0 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="student-access"
>
  <p>
    Choose how students access OpenStax Tutor. Access settings cannot be changed after students begin to enroll.
  </p>
  <a
    href="https://openstax.secure.force.com/help/articles/FAQ/Student-Access-with-Enrollment-Link-versus-LMS-sign-in-for-OpenStax-Tutor-Beta-LMS-Pilots"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-info-circle fa-w-16 c0 ox-icon ox-icon-info-circle"
      data-icon="info-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     Which option is right for my course?
  </a>
  <div>
    <div
      className="links card"
    >
      <div
        className="card-header"
      >
        <div
          aria-selected={false}
          className="links choice"
          onClick={[Function]}
          role="button"
        >
          <div
            className="heading"
          >
            <h3
              className="title"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-square fa-w-14 c0 ox-icon ox-icon-square"
                color="#929292"
                data-icon="square"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Give students direct links
            </h3>
            <p
              className="info"
            >
              You will give students links to access OpenStax Tutor directly.
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <p>
          Send your students their section's direct links to enroll.
        </p>
        <label
          className="copy-on-focus"
        >
          1st
          <input
            className={null}
            onFocus={[Function]}
            readOnly={true}
            value="http://test/period/1/Local-Test-Course-Two-Spring-2017"
          />
        </label>
      </div>
    </div>
    <div
      className="lms card"
    >
      <div
        className="card-header"
      >
        <div
          aria-selected={false}
          className="lms choice"
          onClick={[Function]}
          role="button"
        >
          <div
            className="heading"
          >
            <h3
              className="title"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-square fa-w-14 c0 ox-icon ox-icon-square"
                color="#929292"
                data-icon="square"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Integrate with your learning management system (LMS)
            </h3>
            <p
              className="info"
            >
              Integrate OpenStax Tutor with Blackboard, Canvas, etc., to send student course averages to your LMS and enable single sign on.
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      />
    </div>
  </div>
</div>
`;

exports[`Course Settings, student access renders with only link for non-lms enabled 1`] = `
<div
  className="student-access"
>
  <div
    className="student-access direct-links-only"
  >
    <p>
      Interested in learning management system (LMS) integration? 
      <a
        className="external-icon"
        href="https://openstax.secure.force.com/help/articles/FAQ/LMS-integration-for-OpenStax-Tutor-Beta"
        target="_blank"
      >
        Find out more.
      </a>
    </p>
    <p>
      If not, send your students their section’s direct links (below) to enroll.
    </p>
    <label
      className="copy-on-focus"
    >
      1st
      <input
        className={null}
        onFocus={[Function]}
        readOnly={true}
        value="http://test/period/1/Local-Test-Course-Two-Spring-2017"
      />
    </label>
  </div>
</div>
`;
